include/onnxruntime/core/graph/basic_types.h

```diff
@@ -44,6 +45,6 @@ using IOnnxRuntimeOpSchemaCollectionPtr = std::shared_ptr<IOnnxRuntimeOpSchemaCo
 
 class OpKernel;
 class OpKernelInfo;
-
-using KernelCreateFn = std::function<OpKernel*(const OpKernelInfo& info)>;
+class FuncManager;
+using KernelCreateFn = std::function<onnxruntime::common::Status(FuncManager& func_mgr, const OpKernelInfo& info, std::unique_ptr<OpKernel>& out)>;
```

Comment:
Make it more clear that FuncManager is not const. You can't create a temporary unnamed value of FuncManager type then pass it to  KernelCreateFn.